src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -100,6 +100,71 @@ internal static bool IsValidScopeDesignator(this ExpressionSyntax expression)
             }
         }
 
+        internal static bool IsVariableDeclarationInitialization(this SyntaxNode node)
+        {
+            Debug.Assert(node != null);
+
+            SyntaxNode equalsValueClause = node.Parent;
+
+            if (!equalsValueClause.IsKind(SyntaxKind.EqualsValueClause))
+            {
+                return false;
+            }
+
+            SyntaxNode variableDeclarator = equalsValueClause.Parent;
+
+            if (!variableDeclarator.IsKind(SyntaxKind.VariableDeclarator))
+            {
+                return false;
+            }
+
+            return variableDeclarator.Parent.IsKind(SyntaxKind.VariableDeclaration);
+        }
+
+        /// <summary>
+        /// PROTOTYPE(span):
+        /// short work around until we look into spilling stackalloc expressions
+        /// Because the instruction cannot have any values on the stack before CLR execution.
+        /// Limit it to assignments and conditional expressions for now.
+        /// </summary>
+        internal static bool IsLegalSpanStackAllocPosition(this SyntaxNode node)
+        {
+            Debug.Assert(node != null);
+
+            while (node.Parent.IsKind(SyntaxKind.CastExpression))
+            {
+                node = node.Parent;
+            }
+
+            if (node.Parent.IsKind(SyntaxKind.ConditionalExpression))
+            {
+                node = node.Parent;
+            }
+
+            SyntaxNode parentNode = node.Parent;
+
+            if (parentNode is null)
+            {
+                return false;
+            }
+
+            switch (parentNode.Kind())
+            {
+                case SyntaxKind.EqualsValueClause:
+                    {
+                        // In case of a declaration of a Span<T> variable
+                        return parentNode.Parent.IsKind(SyntaxKind.VariableDeclarator);
+                    }
+                case SyntaxKind.SimpleAssignmentExpression:
```

Comment:
Also this would permit `public int* Prop { get } = stackalloc int[10];`

---
In reply to: [133247521](https://github.com/dotnet/roslyn/pull/21006#discussion_r133247521) [](ancestors = 133247521)